junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/discovery/predicates/IsTestableMethod.java

```diff
@@ -46,6 +49,10 @@ public boolean test(Method candidate) {
 		if (isAbstract(candidate)) {
 			return false;
 		}
+		if (returnsVoid(candidate) != mustReturnVoid) {
```

Comment:
mmh, the more readable version creates a regression here:
`org.junit.jupiter.engine.discovery.predicates.IsTestFactoryMethodTests#factoryMethodReturningVoidEvaluatesToFalse`
which seems to be a valid test case to me...